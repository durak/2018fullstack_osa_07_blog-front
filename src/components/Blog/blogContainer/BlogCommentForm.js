import React from 'react'

const BlogCommentForm = ({ handleComment }) => {

  const valid = (comment) => {
    return comment.length > 0
  }

  const handleSubmit = (event) => {
    event.preventDefault()
    const comment = event.target.comment.value

    if (!valid(comment)) {
      event.target.valid.value = 'required'
      return
    }

    event.target.comment.value = ''
    event.target.valid.value = ''
    handleComment(comment)
  }

  return (
    <div>
      <form onSubmit={handleSubmit} >
        <div>
          <input type="text" name="comment" />
          <output name="valid"></output>
        </div>
        <button type="submit">add comment</button>
      </form>
    </div>
  )
}


export default BlogCommentForm