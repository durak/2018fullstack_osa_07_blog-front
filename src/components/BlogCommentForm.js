import React from 'react'

const BlogCommentForm = ({ handleComment }) => {
  const handleSubmit = (event) => {
    event.preventDefault()
    const comment = event.target.comment.value
    event.target.comment.value = ''

    handleComment(comment)
  }

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <div>
          <input type="text" name="comment" />
        </div>
        <button type="submit">add comment</button>
      </form>
    </div>
  )
}


export default BlogCommentForm