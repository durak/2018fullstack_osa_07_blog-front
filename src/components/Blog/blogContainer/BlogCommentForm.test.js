import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { Form, Button, Input } from 'semantic-ui-react'
import BlogCommentForm from './BlogCommentForm'

describe('<BlogCommentForm />', () => {

  const mockHandleComment = jest.fn()

  it('renders correcly', () => {
    const tree = renderer.create(
      <BlogCommentForm 
        handleComment={mockHandleComment}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot()
  })

/*   it('clicking on submit button calls handler', () => {
    const blogCommentFormComponent = shallow(
      <BlogCommentForm
        handleComment={mockHandleComment}
      />
    )
    const input = blogCommentFormComponent.find('Input')
    input.simulate('focus')
    Input.
    input.simulate('change', { target: { value: 'Changed' } })
    input.simulate('keydown', { 
      which: 'a',
      target: {
        blur() {        
        input.simulate('blur')
      } 
    }
  })
    //input.value = 'valid'
    console.log(input.html())
    const button = blogCommentFormComponent.find('Button')
    button.simulate('click')
    expect(mockHandleComment.mock.calls.length).toBe(1)
  }) */
})