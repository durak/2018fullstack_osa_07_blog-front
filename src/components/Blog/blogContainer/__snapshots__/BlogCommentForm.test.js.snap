// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogCommentForm /> renders correcly 1`] = `
<div>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="inline field"
    >
      <div
        className="ui input"
      >
        <input
          disabled={undefined}
          name="comment"
          onChange={[Function]}
          tabIndex={undefined}
          type="text"
        />
      </div>
      <label>
        <output
          name="valid"
          type="text"
        />
      </label>
    </div>
    <button
      className="ui button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
      type="submit"
    >
      add comment
    </button>
  </form>
</div>
`;
